### PR TITLE
Add docs on how to use new ingress setup

### PR DESCRIPTION
**Pull request by yuvipanda**
Making a new 'advanced' section in the docs for things like this
that need to be documented, but most folks running on public clouds
would not need.

### CODE REVIEW

File: doc/source/advanced.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+[Ingress Controller](https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-controllers) 
+for the ingress object to work. We recommend the
+[nginx ingress](https://github.com/kubernetes/charts/tree/master/stable/nginx-ingress)
+controller maintained by the community.
+
+### Automatic HTTPS with kube-lego & Let's Encrypt
+
+The default automatic HTTPS support does not work if you are using an ingress
+object. You'd have to set it up yourself.
+
```
Comment (choldgraf) [inline doc/source/advanced.md]: two new lines, should be one ya?

File: doc/source/advanced.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+This uses [kube-lego](https://github.com/jetstack/kube-lego) to automatically
+fetch and renew HTTPS certificates
+from [Let's Encrypt](https://letsencrypt.org/). This currently only works with
+the nginx ingress controller & google cloud's ingress controller.
+
+1. Make sure that DNS is properly set up (varies wildly depending on the ingress
+   controller you are using and how your cluster was set up). Accessing
+   `<hostname>` from a browser should route traffic to the hub.
+1. Install & configure kube-lego using the
```
Comment (choldgraf) [inline doc/source/advanced.md]: 2. and next should be 3.

File: doc/source/advanced.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+   [kube-lego helm-chart](https://github.com/kubernetes/charts/tree/master/stable/kube-lego).
+   Remember to change `config.LEGO_EMAIL` and `config.LEGO_URL` at the least.
+2. Add an annotation + TLS config to the ingress so kube-lego knows to get certificates for
+   it:
+   ```yaml
```
Comment (choldgraf) [inline doc/source/advanced.md]: add a newline?

**Comment by yuvipanda** [comment]
@choldgraf changed! 

Yeah, this is mostly for people not running on any of the big public clouds, or people doing somewhat advanced stuff (like the specific install we have on mybinder.org :D)

**Comment by choldgraf** [comment]
sounds good!

**Pull request merged**